proto/agent.go

```diff
@@ -176,9 +176,6 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext) error {
 	if err != nil {
 		return errors.Wrap(err, "exiting due to error marking task complete")
 	}
-	if resp == nil {
```

Comment:
I think this should stay shouldn't it? Otherwise the block below will panic if this does end up happening (unless we want it to panic in this case?)

Reply:
Good catch, thanks. I do, however, want to return nil and not an error so that the agent doesn't exit.